specification.md

```diff
@@ -418,18 +412,49 @@ metadata:
-workflow execution. 
+Function definitions describe services and their operations that need to be invoked during workflow execution.
+They can be referenced by states [action definitions](#Action-Definition)]. 
+
+Note that function definitions focus on defining services which are not invoked via events. 
+To define services that are invoked via events, please reference the [event definitions](#Event-Definition) section, as well as the [actions definitions](#Action-Definition) [eventRef](#EventRef-Definition) property.
+ 
+Because of an overall lack of a common way to describe different services and their operations,
+workflow markups typically chose to define custom function definitions.
+This approach often runs into issues such as lack of markup portability, limited capabilities, as well as 
+forces non-workflow-specific information such as service authentication to be added inside workflow markup.
+
+To avoid these issues, the Serverless Workflow specification mandates that details about 
+services and their operations be described using the [OpenAPI Specification](https://www.openapis.org/) specification.
+OpanAPI is a language-agnostic standard that describes discovery of RESTful services. It is perfectly suited
+for defining every detail about how RESTful services and their operations should be invoked.
+It allows Servlerless Workflow markup to describe RESTful services in a portable 
+way, as well as workflow runtimes to utilize OpenAPI tooling and APIs to invoke those services.
 
-They can be referenced by name in [actions](#Action-Definition) defined in [Event](#Event-State), [Operation](#Operation-State), or [Callback](#Callback-State) workflow states.
+The workflow markup still supports defining non-restful services and their operations using the `metadata` property
+of function definitions. It can however not guarantee its portability on the markup level.
+Runtimes can utilize the function definition `metadata` information to invoke non-restful service operations.
 
-The `resource` property defines the exposed URI of the service that allows its invocation (e.g., via REST).
+The `name` property defines an unique name of the function definition.
 
-The type parameter allows implementations to give more information regarding the function to the readers. 
-It should not affect workflow execution or service invocation. 
+The `operation` property is a combination of the function/service OpenAPI definition document URI and the particular service operation that needs to be onvoked, separated by a '#'. 
+For example `https://petstore.swagger.io/v2/swagger.json#getPetById`. 
+In this example "getPetById" is the OpenAPI ["operationId"](https://swagger.io/docs/specification/paths-and-operations/)
+property in the services OpenAPI definition document which uniquely identifies a specific service operation that needs to be infoked.
+
+The [`metadata`](#Workflow-Metadata) property allows users to define custom information to the custom definitions.
```

Comment:
I think this is probably something we can circle back on later (not blocking this PR), but I struggle a little bit with directing runtimes to use Metadata here as it seems a bit inconsistent with [description of metadata here](https://github.com/serverlessworkflow/specification/blob/master/specification.md#Workflow-Metadata)

> Metadata should not affect workflow execution. Implementations may choose to use metadata information or ignore it. 

I think it depends on peoples idea of what it means to affect workflow execution, but to me specifying what-to-invoke/how-to-invoke-it is something I would interpret to be affecting workflow execution.

I can understand the decision to go this way for now, since we don't have a good standard but this seems to compromise the internal consistency of the doc.  

Some ideas to address this or maybe side-step the inconsistency:
a) Give some reason here for why we're using metadata even though the overall metadata description seems to say not to do this.
b) Relax the description for metadata so this usage of metadata fits
c) What about using a different property name like 'vendor', 'runtime' or something to convey that the attributes under this property are non-portable, user-beware, etc?

I think for the sake of forward progress we should leave it using 'metadata' for now, but I wanted to throw this out there to get peoples thoughts on how we should be thinking about:
a) how to allow for runtimes to model things that affect execution but don't fit well (like the docker examples)...
b) ...while still making it clear to users what parts of their definition/usage are and are-not portable

Reply:
Agree, like the ideas. Will think of something using them. Thanks